samples/Mqtt5/SharedSubscription/src/main/java/sharedsubscription/SharedSubscription.java

```diff
@@ -189,61 +189,43 @@ public static SampleMqtt5Client createMqtt5Client(
 
     public static void main(String[] args) {
 
-        cmdUtils = new CommandLineUtils();
-        cmdUtils.registerProgramName("Mqtt5SharedSubscription");
-        cmdUtils.addCommonMQTTCommands();
-        cmdUtils.addCommonTopicMessageCommands();
-        cmdUtils.registerCommand("key", "<path>", "Path to your key in PEM format. (will use direct MQTT to connect if defined)");
-        cmdUtils.registerCommand("cert", "<path>", "Path to your client certificate in PEM format. (will use direct MQTT to connect if defined)");
-        cmdUtils.registerCommand("client_id", "<int>",
-            "Client id to use (optional, default='test-*'). " +
-            "Note that '1', '2', and '3' will be added for to the given clientIDs since this sample uses 3 clients.");
-        cmdUtils.registerCommand("count", "<int>", "Number of messages to publish (optional, default='10').");
-        cmdUtils.registerCommand("group_identifier", "<string>",
-            "The group identifier to use in the shared subscription (optional, default='java-sample')");
-        cmdUtils.sendArguments(args);
-
-        /* Pull data from the command line */
-        String input_endpoint = cmdUtils.getCommandRequired("endpoint", "");
-        String input_cert = cmdUtils.getCommandRequired("cert", "");
-        String input_key = cmdUtils.getCommandRequired("key", "");
-        String input_ca = cmdUtils.getCommandOrDefault("ca_file", "");
-        String input_clientId = cmdUtils.getCommandOrDefault("client_id", "test-" + UUID.randomUUID().toString());
-        int input_count = Integer.parseInt(cmdUtils.getCommandOrDefault("count", String.valueOf(10)));
-        String input_topic = cmdUtils.getCommandOrDefault("topic", "test/topic");
-        String input_message = cmdUtils.getCommandOrDefault("message", "Hello World!");
-        String input_groupIdentifier = cmdUtils.getCommandOrDefault("group_identifier", "java-sample");
+        /**
+         * cmdData is the arguments/input from the command line placed into a single struct for
+         * use in this sample. This handles all of the command line parsing, validating, etc.
+         * See the Utils/CommandLineUtils for more information.
+         */
+        CommandLineUtils.SampleCommandLineData cmdData = CommandLineUtils.getInputForIoTSample("Mqtt5SharedSubscription", args);
 
         /* If this is CI, append a UUID to the topic */
         if (isCI) {
-            input_topic += "/" + UUID.randomUUID().toString();
+            cmdData.input_topic += "/" + UUID.randomUUID().toString();
```

Comment:
Can we make the appending an UUID to the topic as a default behavior like the client id? Would we have the similar CI problem in other samples? 
```
returnData.input_clientId = getCommandOrDefault(m_cmd_client_id, "test-" + UUID.randomUUID().toString());
```

Reply:
Good idea - I'll move it to the `getCommandOrDefault` for the topic!